### ISSUE TITLE
KeyError during the process (MERSCOPE)

### ISSUE THREAD

**Issue by ykashima2** [gustaveroussy/sopa#133]
Hello Quentin,
I'm trying to use snakemake baysor to analyze our MERSCOPE datasets with the default config file. I have received the following error. Could you give me some advice to solve the problem?

$ snakemake --config data_path=/home/MERSCOPE/S2R3 --configfile=config/merscope/baysor_vizgen.yaml --cores 1 --use-conda

Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job stats:
job               count
--------------  -------
aggregate             1
all                   1
explorer              1
image_write           1
report                1
resolve_baysor        1
total                 6

Select jobs to execute...

[Mon Sep 30 23:44:23 2024]
rule resolve_baysor:
    reason: Missing output files: /home/MERSCOPE/raw_dat/S2R3.zarr/.sopa_cache/baysor_boundaries_done
    resources: tmpdir=/tmp

KeyError: 'CR2d20505a1-64'
Reading transcript-segmentation outputs:   7%|█████████▍                                                                                                                               | 2/29 [00:15<03:30,  7.81s/it]
[Mon Sep 30 23:48:25 2024]
Error in rule resolve_baysor:
    jobid: 2

**Comment by quentinblampey** [comment]
Hello @ykashima2, which version of Baysor are you using? I'm improving support for baysor 0.7.0, so it will fix many issues related to Baysor in the future

**Comment by ykashima2** [comment]
Hi Quentin, Thank you for the reply. I'm using baysor 0.7.0.

**Comment by quentinblampey** [comment]
Alright, it makes sense then. I'll let you know when sopa 2.0.0 is released, it will fix this!

**Comment by quentinblampey** [comment]
Hi @bednarsky, thanks for your kind words, I really appreciate :)

The version 2.0.0 is almost ready, but I'm still adding some tests to make sure it's stable. Also, the new version will use a new feature from the SpatialData library that has not been released yet. So, I will release Sopa 2 when I finish my tests + when the next version of SpatialData is released.

**TL;DR**: let's say something like early/mid-November!

Meanwhile, I'll see if it's easy/quick to cherry-pick my fixes for Baysor on master; I'll let you know.

**Comment by quentinblampey** [comment]
Okay @bednarsky I cherry-picked the change and merged it on `master`, can you `git pull` and let me know if it's fixed?

**Comment by bednarsky** [comment]
Amazing, fixed the issue! Thank you so much! World record time to fix via writing an issue!

**Comment by quentinblampey** [comment]
Nice haha, thanks! I'm closing the issue then
